lib/common/utils.c

```diff
@@ -1077,7 +1077,13 @@ crm_pid_active(long pid, const char *daemon)
     return 0;
 }
 
-#define	LOCKSTRLEN	11
+/* ceil(log10(2^64)) B (for 8 B longs) + 1 B for trailing '\n'
+   Note that while currently Linux on 64 bit architecture cannot be set
+   to support PIDs 2^22+ by default (hence 8 would be enough), it's unclear
+   whether other OS implementations, perhaps with PID space randomization,
+   cannot run into much bigger PIDs so play on the safe side in accordance
+   to existing (unsigned long) getpid result casting. */
+#define	LOCKSTRLEN	21
```

Comment:
The only reason we need LOCKSTRLEN at all is so that crm_read_pidfile() can know whether a file is partially written (everything else would be cleaner with FILE* functions). We already put a newline at the end of the file, so looking for that would probably be a better approach. I'd rather leave this code alone for now than try to make LOCKSTRLEN slightly less hacky.